### PR TITLE
feat: Min reads UMI option to CLI

### PR DESCRIPTION
**Pull request by ivadym**
## Description

> Cancer meeting 2024-06-17:
> - We need to create an argument for UMI settings “1,0,0” or “3,1,1” during config case

#### Added
- Min reads UMI option to CLI

## Documentation

- [x] N/A
- [ ] Updated Balsamic documentation to reflect the changes as needed for this PR.
    - [Document Name]

### Feature Tests

- [x] Test `balsamic-umi` config case
    - Screenshot 2024-06-18 at 17 18 32
- [x] Test `balsamic` config case
    - Screenshot 2024-06-18 at 17 17 41
- [x] `sentieon_consensusfilter_umi` is taking the provided minR value
    - Screenshot 2024-06-19 at 09 33 43

### Pipeline Integrity Tests

- **Report deliver (generation of the `.hk` file)**
    - [x] N/A
    - [ ] Verified
- **TGA T/O Workflow**
    - [x] N/A
    - [ ] Verified
- **TGA T/N Workflow**
    - [x] N/A
    - [ ] Verified
- **UMI T/O Workflow**
    - [x] N/A
    - [ ] Verified
- **UMI T/N Workflow**
    - [ ] N/A
    - [x] Verified
- **WGS T/O Workflow**
    - [x] N/A
    - [ ] Verified
- **WGS T/N Workflow**
    - [x] N/A
    - [ ] Verified
- **QC Workflow**
    - [x] N/A
    - [ ] Verified
- **PON Workflow**
    - [x] N/A
    - [ ] Verified

## Clinical Genomics Stockholm

<details>
<summary></summary>

### Documentation

- **Atlas documentation**
    - [x] N/A
    - [ ] Updated: [Link]
- **Web portal for Clinical Genomics**
    - [x] N/A
    - [ ] Updated: [Link]

### User Changes

- [x] N/A
- [ ] This PR affects the output files or results.
    - [ ] User feedback is considered unnecessary because [Justification].
    - [ ] Affected users have been included in the development process and given a chance to provide feedback.

### Infrastructure Changes

- **Stored files in Housekeeper**
    - [x] N/A
    - [ ] Updated: [Link]
- **CG (CLI and delivered/uploaded files)**
    - [x] N/A
    - [ ] Updated: [Link]
- **Servers (configuration files on Hasta)**
    - [x] N/A
    - [ ] Updated: [Link]
- **Scout interface**
    - [x] N/A
    - [ ] Updated: [Link]

</details>

## Checklist

> [!IMPORTANT]  
> Ensure that all checkboxes below are ticked before merging.

### For Developers

- **PR Description**
    - [x] Provided a comprehensive description of the PR.
    - [x] Linked relevant user stories or issues to the PR.
- **Documentation**
    - [x] Verified and updated documentation if necessary.
- **Tests**
    - [x] Described and tested the functionality addressed in the PR.
    - [x] Ensured integration of the new code with existing workflows.
    - [x] Confirmed that meaningful unit tests were added for the changes introduced.
    - [x] Checked that the PR has successfully passed all relevant code smells and coverage checks.
- **Review**
    - [ ] Addressed and resolved all the feedback provided during the code review process.
    - [ ] Obtained final approval from designated reviewers.

### For Reviewers

- **Code**
    - [x] Code implements the intended features or fixes the reported issue.
    - [x] Code follows the project's coding standards and style guide.
- **Documentation**
    - [x] Pipeline changes are well-documented in the CHANGELOG and relevant documentation.
- **Tests**
    - [x] The author provided a description of their manual testing, including consideration of edge cases and boundary
      conditions where applicable, with satisfactory results.
- **Review**
    - [ ] Confirmed that the developer has addressed all the comments during the code review.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -35,3 +35,4 @@ BALSAMIC_reference/
 BALSAMIC/workflows/umi_workflow
 BALSAMIC/workflows/umi_workflow/.*
 .idea/
+.vscode/
```
Comment (mathiasbio) [inline .gitignore:38]: 👀

Comment (mathiasbio) [reply .gitignore:38]: Don't know if this is a sideeffect from your work environment or just general good practice :D

File: BALSAMIC/workflows/balsamic.smk [hunk 2]
```diff
@@ -134,82 +144,134 @@ fastp_parameters: Dict = get_fastp_parameters(config_model)
 # parse parameters as constants to workflows
 params = BalsamicWorkflowConfig.model_validate(WORKFLOW_PARAMS)
 
+# Custom filter to set the minimum number of reads required to support each UMI tag group
+if config_model.custom_filters and config_model.custom_filters.umi_min_reads:
+    params.umiconsensuscall.filter_minreads = config_model.custom_filters.umi_min_reads
```
Comment (mathiasbio) [inline BALSAMIC/workflows/balsamic.smk:149]: Nice, this should work!

File: tests/utils/test_utils.py [hunk 3]
```diff
@@ -1072,6 +1074,23 @@ def test_validate_cache_version_wrong_format():
         validate_cache_version(click.Context, click.Parameter, cli_version)
 
 
+@pytest.mark.parametrize(
+    "umi_min_reads, should_fail",
+    [("3,2,1", False), ("1,0,0", False), ("3,2", True), ("a,b,c", True), (None, False)],
+)
+def test_validate_umi_min_reads(umi_min_reads: str, should_fail: bool):
```
Comment (mathiasbio) [inline tests/utils/test_utils.py:1081]: That's one good looking test 👌

**Review by mathiasbio — APPROVED** [review]
Looks great to me! I don't see why this shouldn't work. 

In my dummy PR https://github.com/Clinical-Genomics/BALSAMIC/pull/1419/files from the branch where I was testing different UMI-settings, this change if the filter_minreads in the umiconsensuscall settings was all that was required.

File: .gitignore (hunk 1 above)
Comment (ivadym) [reply .gitignore:38]: Oh yes, this is the VS Code workspace settings file (PyCharm uses `.idea`). I've been using VS Code lately, so I added it to the `.gitignore`

**Comment by ivadym** [comment]
> Looks great to me! I don't see why this shouldn't work.
> 
> In my dummy PR https://github.com/Clinical-Genomics/BALSAMIC/pull/1419/files from the branch where I was testing different UMI-settings, this change if the filter_minreads in the umiconsensuscall settings was all that was required.

Thanks for the review @mathiasbio ⭐. Tests have also passed, so I'll merge it into the release v15.0.1 branch and start the verification 😄
